src/testing/testpachd/main.go

```diff
@@ -36,28 +39,51 @@ func main() {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	// Init testpachd options.
-	var opts []pachd.TestPachdOption
-	if *activateAuth {
-		opts = append(opts, pachd.ActivateAuthOption(rootToken))
-	}
-
-	// Build pachd.
+	// Handle cleaning up on exit.
 	ctx, cancel := pctx.Interactive()
+	runCtx := ctx
```

Comment:
Here are the constraints.  

1) We want everything to cleanly shut down when someone presses C-c.  That's pctx.Interactive().  

2) We want to set up pachd and any related services controlled by the cmdline flags.  If Loki is one of those flags, we want the rest of the process to send their logs to Loki, which is done via a zapcore inside the context.  If Loki isn't one of those flags, we don't want to do anything to the context.  

3) We want to clean up when everything is done.  The only way this can really be done is by someone cancelling the root context.  Thus, we need a new background context to handle cleanup, because the original context is cancelled but we don't actually want to cancel cleanup work. 

So this is why there are three contexts.  Building pachd and running pachd are both gated by the Interactive constraint.  That way pachd shuts down cleanly when you request it.   We use the original context to build pachd because it's what's there at the time.  We use runCtx so that if certain OPTIONAL pachd options are specified, pachd.Run() is called with a different context, specifically one that sends all logs from Run to Loki.  This way, `GetLogs` returns pachd logs, just like in the real pachd.  We just happen to implement it through a much different mechanism in production.

Overall, I don't think main is oversized.  It does this: 1) read the options 2) build pachd 3) run pachd in the background 4) connect to pachd and adjust the config file 5) wait for pachd to stop 6) cleanup the ephemeral data that is outside of the process.  That's all it does.